### PR TITLE
[SMALLFIX] Make index definition static

### PR DESCRIPTION
**Pull request by xhad1234**
There is no need to new an index definition every time. Make them static can save some memory.

### CODE REVIEW

File: core/server/src/test/java/alluxio/master/block/BlockMasterPrivateAccess.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
    */
   public static boolean isWorkerRegistered(BlockMaster master, long workerId) {
     IndexedSet<MasterWorkerInfo> workers = Whitebox.getInternalState(master, "mWorkers");
-    IndexDefinition<MasterWorkerInfo> idIndex = Whitebox.getInternalState(master, "mIdIndex");
+    IndexDefinition<MasterWorkerInfo> idIndex = master.ID_INDEX;
```
Comment (apc999) [inline core/server/src/test/java/alluxio/master/block/BlockMasterPrivateAccess.java]: can we still use whitebox to get the static variable? so we could make it still private?

File: core/server/src/main/java/alluxio/master/block/BlockMaster.java [hunk 2]
```diff
@@ -84,6 +84,23 @@
    */
   private static final long CONTAINER_ID_RESERVATION_SIZE = 1000;
 
+  // Worker metadata management.
+  static final IndexDefinition<MasterWorkerInfo> ID_INDEX =
```
Comment (gpang) [inline core/server/src/main/java/alluxio/master/block/BlockMaster.java]: This can still be private?

**Comment by gpang** [comment]
@xhad1234  I left a minor comment about the member visibility. Thanks!

File: core/server/src/test/java/alluxio/master/block/BlockMasterPrivateAccess.java (hunk 1 above)
Comment (jsimsa) [inline core/server/src/test/java/alluxio/master/block/BlockMasterPrivateAccess.java]: it should be possible

**Comment by jsimsa** [comment]
other than resolving the comment about private access to ID_INDEX, this looks good to me

File: core/server/src/test/java/alluxio/master/block/BlockMasterPrivateAccess.java (hunk 1 above)
Comment (xhad1234) [inline core/server/src/test/java/alluxio/master/block/BlockMasterPrivateAccess.java]: Thank you, I changed it to use whitebox.

File: core/server/src/main/java/alluxio/master/block/BlockMaster.java (hunk 2 above)
Comment (xhad1234) [inline core/server/src/main/java/alluxio/master/block/BlockMaster.java]: Yes, I changed as the comment of @apc999.

**Comment by xhad1234** [comment]
Thank you, @gpang @jsimsa @apc999 ,I made some update.

**Comment by jsimsa** [comment]
LGTM

**Comment by apc999** [comment]
LGTM

**Pull request merged**